internal/controlplane/handlers_githubwebhooks.go

```diff
@@ -889,19 +897,23 @@ func (s *Server) processRepositoryEvent(
 
 	l.Info().Msg("handling event for repository")
 
-	dbrepo, err := s.fetchRepo(ctx, event.GetRepo())
+	repoEnt, err := s.fetchRepo(ctx, event.GetRepo())
 	if err != nil {
 		return nil, err
 	}
 
 	// protobufs are our API, so we always execute on these instead of the DB directly.
-	pbRepo := repositories.PBRepositoryFromDB(*dbrepo)
+	pbRepo, err := ghprop.RepoV1FromProperties(repoEnt.Properties)
+	if err != nil {
+		return nil, fmt.Errorf("error converting repository to protobuf: %w", err)
+	}
+
 	eiw := entities.NewEntityInfoWrapper().
 		WithActionEvent(event.GetAction()).
-		WithProjectID(dbrepo.ProjectID).
-		WithProviderID(dbrepo.ProviderID).
+		WithProjectID(repoEnt.Entity.ProjectID).
+		WithProviderID(repoEnt.Entity.ProviderID).
 		WithRepository(pbRepo).
-		WithRepositoryID(dbrepo.ID)
```

Comment:
I wonder if we could/should move away from using the type-specific protobuf messages? On the other hand, they are used by the ingestors and evaluators ...

Reply:
yeah... I would REALLY love to get rid of them. It'll be some work :cry: